File: hw3/hw3_1b.py
# test_gauleg.py  test  gauleg.py  function gaulegf
from gauleg import gaulegf
import math

def f(p):
  return math.sin(p)

print "hw3_1b.py integrate f(p)=sin(x) from 0 to 1"
print "calling x,w = gaulegf(0.0, 1.0, 8)"
n = 8
x,w = gaulegf(0.0, 1.0, n)
print "**************** n =", 8, "*****************"
print "x=",
print x
print "w=",
print w

area = 0.0
for i in range(1, n+1):
  area += w[i]*f(x[i])

print "exact area=1.0-cos(1.0)"
print "computed area= ",
print area
print "error= ",
print math.fabs(area - 1.0 - math.cos(1.0))

n = 16
print "**************** n =", n,  "*****************"
x,w = gaulegf(0.0, 1.0, n)
print "x=",
print x
print "w=",
print w

area = 0.0
for i in range(1, n+1):
  area += w[i]*f(x[i])

print "exact area=1.0-cos(1.0)"
print "computed area= ",
print area
print "error= ",
print math.fabs(area - 1.0 - math.cos(1.0))

File: hw3/hw3_1b.java
public class hw3_1b
{
  public hw3_1b()
  {
    double area;
    double a = 0.0;
    double b = 1.0;
    int n = 8;
    double x[] = new double[n+1];
    double w[] = new double[n+1];

    new gaulegf(a, b, x, w, n);
    area = 0.0;
    for(int i=1; i<=n; i++)
    {
      area += w[i]*f(x[i]);
    }
    System.out.println("Number of points: " + n);
    System.out.println("computed area: " + area);
    System.out.println("error: "+Math.abs((area - (1.0 - Math.cos(1.0)))));

    n = 16;
    x = new double[n+1];
    w = new double[n+1];

    new gaulegf(a, b, x, w, n);
    area = 0.0;
    for(int i=1; i<=n; i++)
    {
	area += w[i]*f(x[i]);
    }
    System.out.println("Number of points: " + n);
    System.out.println("computed area: " + area);
    System.out.println("error: "+Math.abs((area - (1.0 - Math.cos(1.0)))));

  }

  double f(double p) // function to integrate
  {
      return Math.sin(p);
  }

  public static void main (String[] args)
  {
    new hw3_1b();
  }
}